### ISSUE TITLE
Crashing with old/wrong credentials

### ISSUE THREAD

**Issue by redtux** [nextcloud/android#8325]
I just switched to a private profile that I have not been using on my phone for a while, so the saved user password is wrong. Switching to the profile immediately crashes the app.

The bad thing is that I now cannot enter the app any more.

### Steps to reproduce
1. Add profile B to the app and then switch to your default profile A.
2. Change your password server-side (e.g. in your browser or via an external IdM, in my case in LDAP)
3. Back in the app, switch from your default profile A to profile B, and see the app crashing…

### Expected behaviour
- The app should switch from profile A to profile B (with the changed password), 1. asking for the new credentials, or 2. proposing to use a token for the app to avoid having to change the password in the future.

### Actual behaviour
- The app crashes when switching from default profile A to profile B with a new password.

### Environment data
Android version: 9 PKQ1

Device model: Xiaomi Mi MIX 2

Stock or customized system: MIUI 12

Nextcloud app version: 3.16.0 RC2

Nextcloud server version: 19.0.10.1

Reverse proxy: Apache

**Comment by redtux** [comment]
It seems that the app did not forward the backtrace to my bug report, so here it comes:

```
************ CAUSE OF ERROR ************

java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter password
	at okhttp3.Credentials.basic(Unknown Source:8)
	at okhttp3.Credentials.basic$default(Credentials.kt:28)
	at okhttp3.Credentials.basic(Unknown Source:2)
	at com.owncloud.android.lib.common.OwnCloudClientFactory.createNextcloudClient(OwnCloudClientFactory.java:217)
	at com.owncloud.android.operations.RefreshFolderOperation.updatePredefinedStatus(RefreshFolderOperation.java:338)
	at com.owncloud.android.operations.RefreshFolderOperation.updateCapabilities(RefreshFolderOperation.java:311)
	at com.owncloud.android.operations.RefreshFolderOperation.updateOCVersion(RefreshFolderOperation.java:277)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:230)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:359)
	at java.lang.Thread.run(Thread.java:764)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160052
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Xiaomi
Device: chiron
Model: Mi MIX 2
Id: PKQ1.190118.001
Product: chiron

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: V12.0.1.0.PDEMIXM

```

**Comment by redtux** [comment]
@tobiasKaminsky, just to avoid misunderstandings: The account still exists on the remote server, but its password has changed since I last used this profile on my smartphone.

Is there a way to delete the profile in the app so that I can continue using it? E.g., is it save to delete the profile's local folders in Android?

**Comment by redtux** [comment]
@tobiasKaminsky and @AndyScherzinger, thanks a lot for the fix, and congratulations to the new release. Unfortunately, the app is still crashing. Please find the stacktrace below.


```
************ CAUSE OF ERROR ************

java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter password
	at okhttp3.Credentials.basic(Unknown Source:8)
	at okhttp3.Credentials.basic$default(Credentials.kt:28)
	at okhttp3.Credentials.basic(Unknown Source:2)
	at com.owncloud.android.lib.common.OwnCloudClientFactory.createNextcloudClient(OwnCloudClientFactory.java:217)
	at com.owncloud.android.operations.RefreshFolderOperation.updatePredefinedStatus(RefreshFolderOperation.java:338)
	at com.owncloud.android.operations.RefreshFolderOperation.updateCapabilities(RefreshFolderOperation.java:311)
	at com.owncloud.android.operations.RefreshFolderOperation.updateOCVersion(RefreshFolderOperation.java:277)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:230)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:359)
	at java.lang.Thread.run(Thread.java:764)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Xiaomi
Device: chiron
Model: Mi MIX 2
Id: PKQ1.190118.001
Product: chiron

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: V12.0.1.0.PDEMIXM

```

**Comment by redtux** [comment]
Oh, I might have misunderstood… The PR for the bug fix is to be included in 3.17.0, not in 3.16.0, right? If so, sorry for the noise. 🤓
